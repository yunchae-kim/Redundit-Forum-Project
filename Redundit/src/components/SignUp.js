import { useState, useEffect } from "react";
import { useHistory } from "react-router-dom";
import { Button } from "@blueprintjs/core";
import "../style/components/LogInSignUp.css";
import httpClient from "../utility/auth.helper";

const SignUp = () => {
  const history = useHistory();
  const [name, setName] = useState("");
  const [password, setPassword] = useState("");
  const [confirmPass, setConfirm] = useState("");
  const [disabled, setDisabled] = useState(true);
  const [error, setError] = useState(false);

  function handleNameChange(event) {
    setName(event.target.value);
  }

  function handlePasswordChange(event) {
    setPassword(event.target.value);
  }

  function handleConfirmChange(event) {
    setConfirm(event.target.value);
  }

  async function submitNewUser() {
    try {
      const response = await httpClient.signUp(name, password);
      if (response) {
        history.push("/login");
      }
    } catch (err) {
      // eslint-disable-next-line no-alert
      alert("error submitting new user");
    }
  }

  async function handlePress(event) {
    if (event.key === "Enter" && !disabled) {
      submitNewUser();
    }
  }

  useEffect(() => {
    if (name && password && confirmPass && (password === confirmPass)) {
      setError(false);
      setDisabled(false);
    } else if (name && password && confirmPass) {
      setError(true);
      setDisabled(true);
    } else {
      setDisabled(true);
    }
  }, [name, password, confirmPass]);

  return (
    <div className="SU-container">
      <form onSubmit={submitNewUser} className="SU-form">
        <div className="SU-title"> Sign-Up </div>
        <label htmlFor="userEmail" className="SU-entry">
          Username:
          <input
            className="SU-input"
            required
            type="text"
            name="Username"
            placeholder="ex. user12345"
            onChange={handleNameChange}
            onKeyPress={handlePress}
          />
        </label>
        <label htmlFor="userPassword" className="SU-entry">
          Password:
          <input
            className="SU-input"
            required
            type="password"
            name="userPassword"
            placeholder="ex. mypassword"
            onChange={handlePasswordChange}
            onKeyPress={handlePress}
          />
        </label>
        <label htmlFor="userPassword" className="SU-entry">
          Confirm Password:
          <input
            className="SU-input"
            required
            type="password"
            name="userPassword"
            placeholder="confirm password"
            onChange={handleConfirmChange}
            onKeyPress={handlePress}
          />
        </label>
        {error
          ? <p className="errorMessage">Passwords don&lsquo;t match</p>
          : <></>}
        <Button className="SU-submitBtn" text="Sign Up" onClick={() => submitNewUser()} disabled={disabled} />
        <span className="SU-logIn">
          Already have an account? Login
          {" "}
          <a href="login">HERE</a>
          !
        </span>
      </form>
    </div>
  );
};

export default SignUp;
