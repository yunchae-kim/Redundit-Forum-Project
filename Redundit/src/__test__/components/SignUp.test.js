import {
  waitFor,
} from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import SignUp from "../../components/SignUp";

test("test should SignUp match snapshot", async () => {
  const tree = renderer.create(
    <SignUp />,
  ).toJSON();
  await waitFor(async () => {
    expect(tree).toMatchSnapshot();
  });
});
