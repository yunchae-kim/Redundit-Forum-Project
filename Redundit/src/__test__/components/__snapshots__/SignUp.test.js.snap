// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test should SignUp match snapshot 1`] = `
<div
  className="SU-container"
>
  <form
    className="SU-form"
    onSubmit={[Function]}
  >
    <div
      className="SU-title"
    >
       Sign-Up 
    </div>
    <label
      className="SU-entry"
      htmlFor="userEmail"
    >
      Username:
      <input
        className="SU-input"
        name="Username"
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="ex. user12345"
        required={true}
        type="text"
      />
    </label>
    <label
      className="SU-entry"
      htmlFor="userPassword"
    >
      Password:
      <input
        className="SU-input"
        name="userPassword"
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="ex. mypassword"
        required={true}
        type="password"
      />
    </label>
    <label
      className="SU-entry"
      htmlFor="userPassword"
    >
      Confirm Password:
      <input
        className="SU-input"
        name="userPassword"
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="confirm password"
        required={true}
        type="password"
      />
    </label>
    <button
      className="bp3-button bp3-disabled SU-submitBtn"
      disabled={true}
      onBlur={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      tabIndex={-1}
      type="button"
    >
      <span
        className="bp3-button-text"
      >
        Sign Up
      </span>
    </button>
    <span
      className="SU-logIn"
    >
      Already have an account? Login
       
      <a
        href="login"
      >
        HERE
      </a>
      !
    </span>
  </form>
</div>
`;
